controllers/admin/AdminProductsController.php

```diff
@@ -875,6 +875,9 @@ public function processFeatures($id_product = null)
                 $features = isset($form['step1']['features']) ? $form['step1']['features'] : array();
                 if (is_array($features)) {
                     foreach ($features as $feature) {
+                        if (!(int) ($feature['feature'])) {
```

Comment:
You mean, `!is_int($feature['feature'])`?

Reply:
if fact, i mean `if $feature['feature'] is a id_feature present on the shop ` that i have just reduce to   `if (!(int) ($feature['feature']) > 0) ` to avoid perfomance issues ( features are saved on each time the save button is clicked, and i see often merchent having more then 20 fetures per products)

Reply:
You'll see no difference in performance even with more than 100 000 iterations.
After many tests,
`!is_int($var) || empty($var)` is quite faster, more readable and give difference of 0.003ms when iterate with 100 000 data. (Than `!(int)$var`

Reply:
oups sorry, i miss your anwser,

it need to be not 0 to correct the bug described.
as it come from symfony form  features ( https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Form/Admin/Feature/ProductFeature.php#L75), i need to be cas to int ,

i was not clear, maybe we may check if $feature['feature'] is really a correct id_feature to be more clear, whereas it cost  sql's queries.